cargo-shuttle/src/lib.rs

```diff
@@ -950,6 +973,11 @@ impl Shuttle {
             Child,
             RuntimeClient<ClaimService<InjectPropagation<Channel>>>,
         )> = Vec::new();
+
+        if !portpicker::is_free_tcp(run_args.port) {
+            Shuttle::find_available_port(&mut run_args);
```

Comment:
I thought we might need to send in `services.len()` so that it can check the same number of adjacent port numbers as there are services, but I want to double check on those plans. Let me get back on Tuesday with an answer.

Reply:
There is one use case which I encountered, if a port is occupied by another service like NodeJS and if we provide the same port to the **shuttle** then it will start properly but we will get only NodeJS service on that port.

Reply:
But let me check this once again